modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -380,7 +380,20 @@ public ImageFrame[] loadAll(String input, ImageLoadListener listener,
                     }
 
                     if (theStream == null) {
-                        theStream = ImageTools.createInputStream(input);
+                        try {
+                            theStream = ImageTools.createInputStream(input);
+                        } catch (IOException ignored) {
```

Comment:
It looks like this abstracts away errors like `FileNotFoundException`, and it will do a fallback in all cases, even if it is not a 404 or `FileNotFoundException` -- errors like network problems, permission problems, etc, will also trigger a fallback (likely resulting in the same error).

Maybe I'm being too picky :)

Reply:
That is technically solved by re-calling `createInputStream(input)` like @mstr2 suggested to get a correct exception with valid message, so I will go with that approach.

Reply:
This is a bit minor, but what do you think about storing the previous exception in L385, and then re-throwing it? This is most likely exceedingly rare, but it might be possible that the reason why the previous call to `createInputStream` failed was only transient, we're not 100% sure that we can reproduce the exception by attempting it again.

Reply:
This would technically create an ambiguity in the call stack (I checked and it points at in this case L385 instead of where the throw actually happened) but if it's not a problem I can change it.

Reply:
I would think that this is acceptable, since that's the exception message that we expect to see (if the `@1x` version also can't be found).

Reply:
I think the call stack being different is fine; call stacks are there for developers to debug code, and the users that are seeing these will not be debugging that particular code so the callstack being slightly off at the deepest level is fine.  Altenatively, you could still consider adding the `@1x` exception as a supressed exception to the first one (with Exception#addSuppressed`) but I wouldn't bother for this case.

Reply:
All done, with fallback exception added as suppressed